Cargo.toml

```diff
@@ -62,14 +62,17 @@ path = "sdk/core/azure_core_amqp"
 [workspace.dependencies.azure_core_test]
 # azure_core_test is not published and only ever a dev-dependency
 path = "sdk/core/azure_core_test"
+version = "0.1.0"
 
 [workspace.dependencies.azure_core_test_macros]
 # azure_core_test_macros is not published and only ever a dev-dependency
 path = "sdk/core/azure_core_test_macros"
+version = "0.1.0"
 
 [workspace.dependencies.azure_identity]
 # azure_identity should only ever be in dev-dependencies herein
 path = "sdk/identity/azure_identity"
+version = "0.23.0"
```

Comment:
*Ideally*, this is similar to above: this should only ever be a dev-dependency and, ergo, a `path` dependency. A `version` shouldn't be required and it's one less thing to keep up to date. Maybe we exempt this one as well.

Reply:
`azure_identity` is a published package and it's usage as "only dev-dependency" wouldn't be known when looking at updates to make to `workspace.dependencies`

Adding a version where it didn't already exist would only happen in `add` mode where the output wouldn't be committed to the repo.

Reply:
Why not hardcode the exemption, then. It's here only as a dependency and since it should never be anything more than a (transitive) dev-dependency, that's *all* this entry is for.

Reply:
hard coded